### PR TITLE
Wip/web rework

### PR DESCRIPTION
**Pull request by tsundDEV**

### CODE REVIEW

File: apps/web/additional.d.ts [hunk 1]
```diff
@@ -20,5 +20,6 @@ declare namespace NodeJS {
     CLOUDFLARE_TURNSTILE_SECRET: string;
     CLOUDFLARE_TURNSTILE_SITE_KEY: string;
     UPSTASH_REDIS_REST_URL: string;
+    NEXT_PUBLIC_NODE_ENV: string;
```
Comment (ceriddenn) [inline apps/web/additional.d.ts:23]: We should prolly change this to just “NEXT_ENV” sooner than later

**Review by ceriddenn — DISMISSED** [review]

File: apps/web/additional.d.ts (hunk 1 above)
Comment (Huskydog9988) [reply apps/web/additional.d.ts:23]: NODE_ENV is the standard in js, and some packages build differently depending on it. So in my mind NEXT_ENV doesn't provide anything unless next acts differently.

Comment (ceriddenn) [reply apps/web/additional.d.ts:23]: Sorry totally meant NODE_ENV just meant to remove the NEXT_PUBLIC part in front of it

Comment (Huskydog9988) [reply apps/web/additional.d.ts:23]: It's fine, just wanted to make sure I wasn't missing something.

Comment (ceriddenn) [reply apps/web/additional.d.ts:23]: All good

Comment (Huskydog9988) [inline apps/web/additional.d.ts:23]:
```suggestion
    NODE_ENV: string;
```

File: apps/web/server/trpc/router/user.ts [hunk 2]
```diff
@@ -217,12 +216,12 @@ export const userRouter = router({
         try {
           const userCount = await ctx.prisma.user.count({
             where: {
-              role: input.filterRoles,
+              role: input.filterRoles as Roles,
```
Comment (Huskydog9988) [inline apps/web/server/trpc/router/user.ts:219]: is this the only way?

File: apps/web/utils/zod/dash.ts [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
 export const UserSchema = z.object({
   id: z.string(),
-  name: z.string().nullable(),
-  email: z.string().nullable(),
-  emailVerified: z.date().nullable(),
-  image: z.string().nullable(),
+  name: z.string().nullable().optional(),
+  email: z.string().nullable().optional(),
+  emailVerified: z.date().nullable().optional(),
+  image: z.string().nullable().optional(),
```
Comment (Huskydog9988) [inline apps/web/utils/zod/dash.ts:8]: doesn't optional and nullable in zod do the same thing?

File: docs/src/config.ts [hunk 4]
```diff
@@ -8,7 +6,7 @@ export const SITE = {
 
 export const OPEN_GRAPH = {
   image: {
-    src: 'https://raw.githubusercontent.com/waldo-vision/waldo/master/apps/web/public/android-chrome-256x256.png',
+    src: '../public/android-chrome-256x256.png',
```
Comment (Huskydog9988) [inline docs/src/config.ts:9]: does this work?

File: docs/src/pages/en/introduction.md [hunk 5]
```diff
@@ -10,6 +10,6 @@ This is the documentation for all of the Waldo project. If your unaware of what
 
 ## Getting Started
 
-To start contributing to Waldo, check our [getting started](getting-started) page. It provides all the documentation on how to setup Waldo so you can start contributing.
+To start contributing to Waldo, check our [getting started](/en/getting-started) page. It provides all the documentation on how to setup Waldo so you can start contributing.
```
Comment (Huskydog9988) [inline docs/src/pages/en/introduction.md:13]: getting started should work, and keeps it a realative path

File: turbo.json [hunk 6]
```diff
@@ -13,15 +13,16 @@
...
         "BTLNET_CLIENT_ID",
         "BTLNET_CLIENT_SECRET",
         "FB_CLIENT_ID",
         "FB_CLIENT_SECRET",
         "TWITCH_CLIENT_ID",
         "TWITCH_CLIENT_SECRET",
         "UPSTASH_REDIS_REST_URL",
-        "UPSTASH_REDIS_REST_TOKEN"
+        "UPSTASH_REDIS_REST_TOKEN",
+        "NEXT_PUBLIC_NODE_ENV"
```
Comment (Huskydog9988) [inline turbo.json:25]:
```suggestion
        "NODE_ENV"
```

**Review by Huskydog9988 — CHANGES_REQUESTED** [review]

**Review by jaredb1011 — APPROVED** [review]

File: docker/cockroach/.gitignore [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore
```
Comment (Huskydog9988) [inline docker/cockroach/.gitignore:4]: why not just use the root git ignore?

**Review by Huskydog9988 — APPROVED** [review]

File: docker/cockroach/.gitignore (hunk 7 above)
Comment (tsundDEV) [reply docker/cockroach/.gitignore:4]: I want to have this directory showing on the git

**Review by ceriddenn — APPROVED** [review]

**Pull request merged**